.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -24,6 +24,16 @@ Enter your extended release note in the block below.
 
 #### Additional Notes for your reviewer:
 
+##### Review Checklist:
+
+- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
+- [ ] Relevant tests are added or updated
+- [ ] Relevant internal docs added or updated
```

Comment:
What is meant by internal here? If it's VMware specific, I would prefer to leave it out.